crates/host/src/wasmbus/mod.rs

```diff
@@ -1255,77 +1252,91 @@ impl Host {
             component,
             id,
             handler,
-            exports: spawn(
-                async move {
-                    join!(
-                        async move {
-                            let mut tasks = JoinSet::new();
-                            let mut exports = stream::select_all(exports);
-                            loop {
-                                let permits = Arc::clone(&permits);
-                                select! {
-                                    Some(fut) = exports.next() => {
-                                        match fut {
-                                            Ok(fut) => {
-                                                debug!("accepted invocation, acquiring permit");
-                                                let permit = permits.acquire_owned().await;
-                                                tasks.spawn(async move {
-                                                    let _permit = permit;
-                                                    debug!("handling invocation");
-                                                    match fut.await {
-                                                        Ok(()) => {
-                                                            debug!("successfully handled invocation");
-                                                            Ok(())
-                                                        },
-                                                        Err(err) => {
-                                                            warn!(?err, "failed to handle invocation");
-                                                            Err(err)
-                                                        },
-                                                    }
-                                                });
-                                            }
-                                            Err(err) => {
-                                                warn!(?err, "failed to accept invocation")
-                                            }
+            exports: spawn(async move {
+                join!(
+                    async move {
+                        let mut tasks = JoinSet::new();
+                        let mut exports = stream::select_all(exports);
+                        loop {
+                            let permits = Arc::clone(&permits);
+                            select! {
+                                Some(fut) = exports.next() => {
```

Comment:
Like below, looks like we should be able to instrument here just once instead of the duplication in the runtime crate

Reply:
I do know that instrumenting the `exports.next()` future here won't work, since that gets polled continuously until a request comes in. But I did attempt to instrument the `fut`, and I remembered that being disconnected. Getting back into investigating this today